patches/spec.fetched.json/16-findHWReservationById-response-schema-devicecreateinput-customdata-default.patch

```diff
@@ -0,0 +1,50 @@
+diff --git a/spec/oas3.patched.json b/spec/oas3.patched.json
+index aff92d5..8ed228d 100644
+--- a/spec/oas3.patched.json
++++ b/spec/oas3.patched.json
+@@ -1001,21 +1001,21 @@
+               "enum": [
+                 "hourly",
+                 "daily",
+                 "monthly",
+                 "yearly"
+               ],
+               "type": "string"
+             },
+             "customdata": {
+               "description": "Customdata is an arbitrary JSON value that can be accessed via the\nmetadata service.",
+-              "default": {}
++              "default": null
```

Comment:
The backend allows for this field to be submitted as a JSON string (`"customdata": "{\"foo\": \"bar\"}"`) or as an object (`"customdata": {"foo":"bar"}`).

When omitted in the request, the default value in the response is : `{}` (an empty object).

My inclination here would be to represent the field as an object with default `{}`. Is the problem that `type: object` was not included in the OAS3 spec?

```suggestion
              "default": {},
              "type": "object",
              "additionalProperties": true,
```

Reply:
Interesting thing to note is the older generator assigned customdata=null. This behaviour changed after version bump. Refer `equinix-openapi-metal/.openapi-generator/VERSION`

Reply:
> The backend allows for this field to be submitted as a JSON string (`"customdata": "{\"foo\": \"bar\"}"`) or as an object (`"customdata": {"foo":"bar"}`).

So with this change, customdata will have one string key `"{\"foo\": \"bar\"}"` mapped to `null` or an empty string `""` mapped to` {"foo":"bar"}` after deserialization?

Reply:
`customdata` is the key to an arbitrary value. You may submit a string, null, a boolean, or an object here.

However, if the request looks like quoted JSON, it will be parsed.  See the "serialized in" columns of this table: https://github.com/packethost/packngo/pull/225#issuecomment-744594881

This is why I believe the best OAS3 approach will be `additionalProperties: true`, with `type: object`.

Reply:
Makes sense, incorporated suggestion in https://github.com/equinix-labs/metal-java/pull/52/commits/2587209657caebfaa20150b9948dd80eb1689f8e